### PR TITLE
feat(chart): introduce global tolerations and nodeselectors for user deployed components

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: https://github.com/longhorn/longhorn/issues/5089

- Add global tolerations and nodeselectors for user deployed components
- Use global setting by default
- Override each by their specific settings .

### CODE REVIEW

**Review by ejweber — DISMISSED** [review]
LGTM.

**Comment by innobead** [comment]
@jillian-maroket Please help review this. Thanks.

File: chart/README.md [hunk 1]
```diff
@@ -79,6 +79,8 @@ The `values.yaml` contains items used to tweak a deployment of this chart.
 | global.cattle.windowsCluster.enabled | bool | `false` | Setting that allows Longhorn to run on a Rancher Windows cluster. |
 | global.cattle.windowsCluster.nodeSelector | object | `{"kubernetes.io/os":"linux"}` | Node selector for Linux nodes that can run user-deployed Longhorn components. |
 | global.cattle.windowsCluster.tolerations | list | `[{"effect":"NoSchedule","key":"cattle.io/os","operator":"Equal","value":"linux"}]` | Toleration for Linux nodes that can run user-deployed Longhorn components. |
+| global.nodeSelector | object | `{}` | Node selector for nodes that can run user-deployed Longhorn components (Manager, UI, Driver Deployer). |
```
Comment (jillian-maroket) [inline chart/README.md:82]:
```suggestion
| global.nodeSelector | object | `{}` | Node selector for nodes allowed to run user-deployed components such as Longhorn Manager, Longhorn UI, and Longhorn Driver Deployer. |
```

File: chart/README.md [hunk 2]
```diff
@@ -79,6 +79,8 @@ The `values.yaml` contains items used to tweak a deployment of this chart.
...
+| global.tolerations | list | `[]` | Toleration for nodes that can run user-deployed Longhorn components (Manager, UI, Driver Deployer). |
```
Comment (jillian-maroket) [inline chart/README.md:83]:
```suggestion
| global.tolerations | list | `[]` | Toleration for nodes allowed to run user-deployed components such as Longhorn Manager, Longhorn UI, and Longhorn Driver Deployer. |
```

File: chart/values.yaml [hunk 3]
```diff
@@ -2,6 +2,10 @@
 # This is a YAML-formatted file.
 # Declare variables to be passed into your templates.
 global:
+  # -- Toleration for nodes that can run user-deployed Longhorn components (Manager, UI, Driver Deployer).
```
Comment (jillian-maroket) [inline chart/values.yaml:5]:
```suggestion
  # -- Toleration for nodes allowed to run user-deployed components such as Longhorn Manager, Longhorn UI, and Longhorn Driver Deployer.
```

File: chart/values.yaml [hunk 4]
```diff
@@ -2,6 +2,10 @@
...
+  tolerations: []
+  # -- Node selector for nodes that can run user-deployed Longhorn components (Manager, UI, Driver Deployer).
```
Comment (jillian-maroket) [inline chart/values.yaml:7]:
```suggestion
  # -- Node selector for nodes allowed to run user-deployed components such as Longhorn Manager, Longhorn UI, and Longhorn Driver Deployer.
```

**Review by jillian-maroket — COMMENTED** [review]
@ChanYiLin Sorry for the delay. Please check.

**Comment by ChanYiLin** [comment]
Hi @jillian-maroket 
Sorry for replying late, I have fixed the doc. Thanks

**Review by jillian-maroket — APPROVED** [review]
@ChanYiLin LGTM. Thanks!

**Comment by ChanYiLin** [comment]
Hi @derekbit 
I am not sure, but I think we can keep the flexibility

**Review by innobead — APPROVED** [review]

**Comment by ChanYiLin** [comment]
@mergify backport v1.6.x

**Comment by ChanYiLin** [comment]
@mergify backport v1.5.x

**Pull request merged**